plugins/source.go

```diff
@@ -106,25 +102,25 @@ func (p *SourcePlugin) Version() string {
 
 // Sync is syncing data from the requested tables in spec to the given channel
 func (p *SourcePlugin) Sync(ctx context.Context, logger zerolog.Logger, spec specs.Source, res chan<- *schema.Resource) (*schema.SyncSummary, error) {
-	c, err := p.newExecutionClient(ctx, logger, spec)
+	spec.SetDefaults()
+	if err := spec.Validate(); err != nil {
+		return nil, fmt.Errorf("invalid spec: %w", err)
+	}
+	tableNames, err := p.listAndValidateTables(spec.Tables, spec.SkipTables)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create execution client for source plugin %s: %w", p.name, err)
+		return nil, err
 	}
 
-	// limiter used to limit the amount of resources fetched concurrently
-	concurrency := spec.Concurrency
-	if concurrency == 0 {
-		concurrency = defaultConcurrency
+	c, err := p.newExecutionClient(ctx, logger, spec)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create execution client for source plugin %s: %w", p.name, err)
 	}
-	logger.Info().Uint64("concurrency", concurrency).Msg("starting source plugin sync")
-	goroutinesSem := semaphore.NewWeighted(helpers.Uint64ToInt64(concurrency))
+	logger.Info().Interface("spec", spec).Msg("starting sync")
+	goroutinesSem := semaphore.NewWeighted(helpers.Uint64ToInt64(spec.TableConcurrency))
```

Comment:
Should we perhaps rename `goroutinesSem` to `tablesSem` for consistency?

Reply:
Fixed